cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/app/etl/batch/ETLMapReduce.java

```diff
@@ -141,6 +188,16 @@ public void beforeSubmit(MapReduceContext context) throws Exception {
     job.setNumReduceTasks(0);
   }
 
+  private void addPropertiesToErrorDataset(String errorDatasetName, MapReduceContext context) {
+    if (errorDatasetName != null) {
```

Comment:
nit: a little better to check this before calling this method, as it's more consistent with style elsewhere